server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -74,23 +76,22 @@
  */
 @PublicApi(since = "1.0.0")
 public class NodeIndicesStats implements Writeable, ToXContentFragment {
-    private CommonStats stats;
-    private Map<Index, List<IndexShardStats>> statsByShard;
+    protected CommonStats stats;
+    protected Map<Index, CommonStats> statsByIndex;
```

Comment:
We could but my concern was more in the line of version upgrade scenarios, when the response does not contain this aggregationLevel as nodes with prior version will not contain it.

Reply:
This is why I am resolving this information dynamically as the stream input is coming in. The logic is as follows:

1. If the response is coming from a previous version node, it will only have statsByShards and so we just backfill it and accordingly generate the XContent later.
2. If the response is coming from a current version node or later, it may contain statsByIndex. In this case, we check for it and act accordingly when generating the XContent.